In [26]:
str1 = "Sentences work too"
str2 = "Sentences wAke too"

In [31]:
def hamming(str1, str2):
    dist = 0
    count = 0
    for a, b in zip(str1,str2):
        if (a != b):
            if (a.isupper() and b.islower() and count != 0):
                dist += 0.5
            if (a.islower() and b.isupper() and count != 0):
                dist += 0.5
            if (a.upper() != b.upper()):
                if(a.upper == "Z" and b.upper == "S"):
                    dist+=0
                elif(a.upper == "S" and b.upper == "Z"):
                    dist+= 0
                else:
                    dist+=1
        count +=1
    print(dist)
        

In [27]:
hamming(str1,str2)

3.5


## 2.a

In [28]:
a = "data Science"
b = "Data Sciency"
hamming(a,b)

1


## 2.b

In [32]:
a = "organizing"
b= "orGanising"
hamming(a,b)

1.5


#### I understand the above should be 0.5 instead of 1.5. The program is having difficulty considering 's' to be the same as 'z,' but I do not understand why

## 2.c

In [33]:
a = "AGPRklafsdyweIllIIgEnXuTggzF"
b = "AgpRkliFZdiweIllIIgENXUTygSF"
hamming(a,b)

8.5


#### Again, this should be 6.5 instead of 8.5

## 2.d Scenario: a Hamming distance algorithm would be useful in providing user suggestions. If they entered an unrecognized word into an interface (they made a typo), the Hamming distance algorithm could be used to survey the viable input options and compare the scores for each option. The lower the Hamming score, the more likely the user intended to select the given option. The program could then return the option with the lowest Hamming score, saying, "Did you mean [x]?"